packages/react-api/src/hooks/useCartoLayerProps.js

```diff
@@ -29,12 +31,16 @@ export default function useCartoLayerProps({
     viewport,
     spatialFilter,
     uniqueIdProperty,
-    debounceTimeout: viewporFeaturesDebounceTimeout
+    debounceTimeout: viewporFeaturesDebounceTimeout,
+    spatialIndex
   });
 
   const onDataLoad = useCallback(
     (data) => {
       if (data?.tilejson) {
+        setSpatialIndex(
+          Object.values(SpatialIndex).includes(data.scheme) ? data.scheme : undefined
+        );
         return onDataLoadForTile(data);
```

Comment:
Since the spatialIndex state is only used in useTileFeatures, I think we should move that state inside the hook and move 41-43 lines inside the onDataLoadForTile. What do you think?

Reply:
Totally agree